eZ/Publish/Core/Search/Legacy/Content/Mapper/FullTextMapper.php

```diff
@@ -101,37 +101,39 @@ protected function getFullTextValues(Content $content)
                     'isMainAndAlwaysAvailable' => (
                         $field->languageCode === $contentInfo->mainLanguageCode && $contentInfo->alwaysAvailable
                     ),
+                    'transformationRules' => $transformationRules,
+                    'splitFlag' => $splitFlag,
                 ]
             );
         }
 
         return $fullTextValues;
     }
 
-    /**
-     * Get FullTextField value.
-     *
-     * @param Content\Field $field
-     * @param Type\FieldDefinition $fieldDefinition
-     *
-     * @return string
-     */
-    private function getFullTextFieldValue(Content\Field $field, Type\FieldDefinition $fieldDefinition)
+    private function getFullTextFieldData(Content\Field $field, Type\FieldDefinition $fieldDefinition): array
```

Comment:
I consider it a bit of anti-pattern to return multiple loosely related data from a method for the sake of not "inlining" the code itself. I mean, the array here will always have 3 elements that should be treated separately.
What I would do instead, to avoid direct inline and have somewhat more consistent style, I'd move some parts around and re-design this internal method to just return Field

Suggested diff: https://github.com/ezsystems/ezpublish-kernel/compare/EZP-31287...alongosz:review/ezp-31287-email-isbn-indexing. See that applying  the diff results in less code and drops redundancy of `!is_array : implode` expression (used twice before).

The alternative would be just to pack them in some sort of DTO object instead of non-strict array.

Reply:
@alongosz  applied your solution in https://github.com/ezsystems/ezpublish-kernel/pull/3023/commits/0e16ccad98a970790d341e6ac610dc6a43b11aa7